addons/stock/product.py

```diff
@@ -287,6 +287,19 @@ def action_view_routes(self, cr, uid, ids, context=None):
         template_obj = self.pool.get("product.template")
         templ_ids = list(set([x.product_tmpl_id.id for x in self.browse(cr, uid, ids, context=context)]))
         return template_obj.action_view_routes(cr, uid, templ_ids, context=context)
+    
+    def onchange_track_all(self, cr, uid, ids, track_all, context=None):
+        warning = {}
+        if ids and track_all:
+            quant_ids = self.pool.get('stock.quant').search(cr, uid, [('product_id','=',ids), ('lot_id','=', False), ('location_id.usage','=', 'internal')])
```

Comment:
Here you should call the function for the related template as the track_all is defined on product template level.  self.pool.get('product.template').onchange_track_all (cr, uid, [...  (ids of product_tmpl_id)